backend/src/user/service.ts

```diff
@@ -38,8 +39,16 @@ export const processUserAuthAttempt = async (provider: Provider, profile: passpo
     }
 
     try {
-        const id = await saveUser(user);
-        return { ...user, _id: id } as User;
+        const doesUserExist = await getUserByUuid(user.uuid);
+        if (!doesUserExist) {
+            user.role = Role.USER;
+            const id = await saveUser(user);
```

Comment:
Powtórzenie kodu. Można to przenieść poza ifa
const id = await saveUser(user);
return { ...user, _id: id } as User;